src/main/java/org/minions/devfund/carlosorellana/Squarelotron.java

```diff
@@ -0,0 +1,276 @@
+            }
+        }
+        return squarelotronTemporal;
+    }
+
+    /**
+     * Creates a new matrix to Main Diagonal Flip values.
+     *
+     * @param positionX            - Position X.
+     * @param positionY            - Position Y.
+     * @param first                - Value get to (ring - 1).
+     * @param last                 - Value get to (size - ring).
+     * @param squarelotronTemporal - New Squarelotron Object.
+     * @return Instance of Squarelotron.
+     */
+    private Squarelotron createNewMatrixToMainDiagonalFlip(final int positionX, final int positionY,
+                                                           final int first, final int last,
+                                                           final Squarelotron squarelotronTemporal) {
+        boolean valA = positionX == first || positionX == last;
+        boolean valB = positionY >= first && positionY <= last;
+        boolean valC = positionX > first && positionX < last;
+        boolean valD = positionY == first || positionY == last;
+
+        if (valA && valB) {
+            squarelotronTemporal.squarelotron[positionY][positionX] = this.squarelotron[positionX][positionY];
+        }
+
+        if (valC && valD) {
+            squarelotronTemporal.squarelotron[positionY][positionX] = this.squarelotron[positionX][positionY];
+        }
+        return squarelotronTemporal;
+    }
+
+    /**
+     * Rotates Matrix 90 grades positive or negative position.
+     *
+     * @param numberOfTurns Number or rotations.
+     */
+    public void rotateRight(final int numberOfTurns) {
+
+        int turns = numberOfTurns;
+        int ring;
+        boolean right = true;
+        int maxRing = getMaxRings();
+
+        // New Temporal Squarelotron
+        int[][] temporalSquarelotron = new int[size][size];
+        temporalSquarelotron = duplicateSquarelotronMatrix(temporalSquarelotron);
+
+
+        if (numberOfTurns < 0) {
+            right = false;
+            turns = turns * (-1);
+        }
+
+        for (int x = 0; x < turns; x++) { // Move the ring in base the number of turns.
+            for (int y = 1; y <= maxRing; y++) { // Mode on the ring selected
+                ring = y;
+
+                for (int i = 0; i < size; i++) {
+                    for (int j = 0; j < size; j++) {
+
+                        createNewMatrixToRotateRight(right, temporalSquarelotron, ring, i, j);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Updates the original Matrix with the values assigned to rotate the matrix.
+     *
+     * @param right - Directions to rotate the matrix.
+     * @param temporalSquarelotron - Clone Squarelotron Matrix.
+     * @param ring - Number ring.
+     * @param i - Position i.
+     * @param j - Position j.
+     */
+    private void createNewMatrixToRotateRight(final boolean right, final int[][] temporalSquarelotron,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)